src/NUnitFramework/tests/Api/FrameworkControllerTests.cs

```diff
@@ -76,13 +80,25 @@ public void TestOutOfFixtureCategory()
             }
         }
 
+        [OneTimeSetUp]
+        public void OneTimeSetup()
+        {
+            _originalPrefix = Test.IdPrefix;
+        }
+
         [SetUp]
         public void CreateController()
         {
-            _controller = new FrameworkController(MOCK_ASSEMBLY_PATH, "ID", _settings);
+            _controller = new FrameworkController(MOCK_ASSEMBLY_PATH, Test.IdPrefix, _settings);
             _handler = new CallbackEventHandler();
         }
 
+        [OneTimeTearDown]
+        public void TearDown()
+        {
+            Test.IdPrefix = _originalPrefix;
```

Comment:
Is there any scenario in which `Test.IdPrefix` has been changed, the finally blocks do not execute, and `TearDown` can run?

Reply:
I don't think so, but I'd rather be safe than sorry, which is why I added it. It should be able to be removed safely though.

Reply:
Its presence here implies that it is doing something. Would you be willing to remove it?